elastic/kibana
# AI Assistant Management Plugin + Knowledge Base Management

## Summary

This PR adds a bunch of plugins to help manage AI Assistant Management settings.

It offers a 'selection' plugin inside Stack Management where a user can select which AI Assistant she wants to manage.
The Security team can hook into this one, so settings for both AI Assistants can be accessed from inside one place inside Stack Management.

This PR also adds the plugin to manage settings for the AI Assistant for Observability, including Knowledge Base management. This plugin is available both in Stack Management (stateful) and Project Settings (serverless). 

## What it looks like
https://github.com/elastic/kibana/assets/535564/51392ec5-05c9-4947-9bf2-810d8d0b7525


## Detailed
1. **Adds a Stack Management plugin** (`/src/plugins/ai_assistant_management/selection`). Its primary function is to render a selection screen to help users navigate to the settings plugin for the AI Assistant for a specific solution. This plugin is displayed in Stack Management, which is only available in stateful versions of Kibana.

2. **Adds a AI Assistant for Observability Settings plugin** (`/src/plugins/ai_assistant_management/observability`). This plugin allows management of specific Observability AI Assistant settings. It is available in stateful versions of Kibana (via the aforementioned Stack Management plugin) or in serverless versions via Project Management.

3. **Knowledge Base management for Observability AI Assistant**: The AI Assistant for Observability Settings plugin has a Knowledge Base tab, which allows users to add / read / update / delete and bulk import entries into the Knowledge Base of the Observability AI Assistant.

4. **Moving of KB endpoints in Observability AI Assistant plugin**: KB endpoints and functions were located in the same folder. As this PR adds new endpoints for the KB for CRUD operations, it also moves the existing ones from the function folder into a dedicated one so there's a clearer distinction between kb and functions.

5. **Adding of GenAI Connector inside Chat Flyout**: If the user has admin rights, it is possible to set up a GenAI connector from within the Observability AI Assistant Chat Flyout. This provides a faster and more seamless onboarding experience. If the user does not, she will be redirected to the Settings page.

## Bug fixes
* Fixes chat item styling issues (padding, background color).

## How to test
* Check if the Stack Management plugin works on stateful
* Check if the AI Assistant Settings plugin works on stateful + serverless
* Check if CRUD operations on KB work
* Check if searching on KB entries work
* Check if its possible to navigate to KB tab directly (`app/management/kibana/aiAssistantManagementObservability?tab=knowledge_base`)


## Todo
- [x] Add sorting to getEntries
- [x] Add params for tab routing
- [x] Add API tests
- [ ] Add fallback for already indexed entries when searching
